### PR TITLE
Correctness and performance improvements to miniMD

### PR DESCRIPTION
**Pull request by benharsh**
Correctness improvements:
- StencilArr.dsiBoundaries() now yields all outer boundary chunks with the correct global offset coordinates
- StencilArr.dsiBoundaries() is now a standalone parallel iterator
- The readRemote() function has been retired in favor of reading from a slice into the non-boundary regions of the array

Performance improvements:
- More parallelism in dsiUpdateFluff and dsiBoundaries
- reduced communication in force computation step

### CODE REVIEW

**Comment by benharsh** [comment]
Passes gasnet testing for the few tests that actually use StencilDist (miniMD, stencil9, PRK)

**Comment by ben-albrecht** [comment]
Took me a bit, but I think I understand this now. It looks good to me, and I have no suggested changes at the moment. It might be worth getting a second opinion on it if @bradcray is available.

File: test/release/examples/benchmarks/miniMD/helpers/neighbor.chpl [hunk 1]
```diff
@@ -121,8 +119,9 @@ proc buildNeighbors() {
 // See test/types/records/bharshbarger/remote*Record.chpl for more.
 inline proc addatom(ref a : atom, x : v3, b : v3int) {
   // increment bin's # of atoms
-  Count[b] += 1;
-  const end = if useStencilDist then Count.readRemote(b) else Count[b];
+  var end = RealCount[b];
+  end += 1;
```
Comment (vasslitvinov) [inline test/release/examples/benchmarks/miniMD/helpers/neighbor.chpl]: or could say:  const end = RealCount[b] + 1;

File: test/release/examples/benchmarks/miniMD/helpers/StencilDist.chpl [hunk 2]
```diff
@@ -1248,40 +1234,87 @@ iter _array.boundaries() {
...
-iter _array.boundaries(param tag : iterKind, followThis) where tag == iterKind.follower {
-  yield followThis;
-}
-
 iter StencilArr.dsiBoundaries() {
   for i in dom.dist.targetLocDom {
-    for (D, N, L) in zip(locArr[i].Dest, locArr[i].Neighs, locArr[i].NeighDom) {
-      if (i + L != N) {
-        for el in locArr[i].myElems[D] do yield (el, L);
+    var loc = locArr[i];
```
Comment (vasslitvinov) [inline test/release/examples/benchmarks/miniMD/helpers/StencilDist.chpl]: The name 'loc' is a bit confusing. How about change to 'lsa' or 'locDesc' ?

**Comment by vasslitvinov** [comment]
OK to merge as-is.
I suggested a couple of minor mods - they are optional.

**Pull request merged**